src/consumer/consumerGroup.js

```diff
@@ -110,6 +115,7 @@ module.exports = class ConsumerGroup {
       }
 
       await this.cluster.refreshMetadataIfNecessary()
```

Comment:
Maybe change this to always refresh metadata if you are the leader, so we don't need to do another round of re-syncs if the leader is out of date.